source/_posts/durable-entities.md

```diff
@@ -0,0 +1,129 @@
+    {
+        var bedNumber = req.Query["bedNumber"];
+        var entityId = new EntityId(nameof(BedEntity),bedNumber);
+        
+        log.LogInformation($"Received request for bed {bedNumber}");
+        
+        var bedEntity = await durableEntityClient.ReadEntityStateAsync<BedEntity>(entityId);
+        if (bedEntity.EntityExists && bedEntity.EntityState.IsOccupied)
+        {
+            return new BadRequestObjectResult("Bed is already occupied.");
+        }
+        
+        await durableEntityClient.SignalEntityAsync(entityId, nameof(BedEntity.AssignBedAsync));
+
+        return new OkObjectResult($"Bed {bedNumber} has been set to occupied.");
+    }
+}
+```
+
+If we now run this client Function and send a request to it (bedNumber = 123) with a breakpoint on the logline we see the entityId being constructed as `@BedEntity@123`. The framework will use this id to get the current state if an entity is found using this id. If an entity is found using the `ReadEntityStateAsync` method on the `IDurableEntityClient` we check if the bed is also occupied. If that is the case we cannot assign it and return a `BadRequest`. If no BedEntity could be found with this id or the bed is not occupied we signal the entity to assign a bed with the provided `bedNumber` using the `SignalEntityAsync` method. This method gets the entity and triggers the `AssignBedAsync` call which changes the state. The framework then persists the changes in the entity state back to table storage.
+
+But what happens in the underlaying framework if we call `SignalEntityAsync`? What happens is that the framework starts up a virtual actor as Microsoft calls it. Its a function that acts on our request to process information regarding the entities state. For the communication, the framework uses the existing durable queues that we know from the already existing durable functions framework which this is built on.
+
+Durable entities are in no way replacing the previously existing durable functions. They are an addition to the existing durable function framework to help you to get rid of some of the plumbing work that was still there. You can use all the parts that were already there in combination with this new addition. 
+
+![Durable Function Types](images/durable-entities/functiontypes.png)
+
+One of the key parts that Entities resolve is if you would send a lot of `AssignBed` requests the framework would process them for the entity there are requesting one by one. This means you don't need to write locks or checks to validate the concurrency. This is all handled by the framework.
+
+In our sample, we only used the Client function and the Entity to get the job done. If you need more flexibility you could easily add an Orchestrator function that we know from the existing durable functions to the mix that handles more complex scenarios.
+
+The complete code for the example shown previously can be found [here](https://github.com/fschaal/Durable-Entities).
```

Comment:
nice! gewoon alle code, klasse! deze zou ik onder de intro zetten. waar je ook vermeld: 'we will be using an hospital application'